libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -141,6 +148,23 @@ private static ElasticsearchEntitlementChecker initChecker() {
         }
     }
 
+    private static Class<?> getVersionSpecificCheckerClass(int javaVersion, String packageName, String baseClassName) {
+        final String classNamePrefix;
+        if (javaVersion >= 23) {
+            classNamePrefix = "Java23";
+        } else {
+            classNamePrefix = "";
+        }
```

Comment:
This prefix logic might need an explanatory comment. It looks a bit random at first glance.